posthog/api/insight.py

```diff
@@ -260,6 +259,9 @@ def to_representation(self, instance):
             filters = instance.dashboard_filters()
             representation["filters"] = filters
 
+        # upgrade the query to the latest version
+        representation["query"] = upgrade(representation["query"])
```

Comment:
should we also call `upgrade` in the `InsightSerializer.update(...)` or `valiate_query` method so every time an insight is edited, we save the upgraded version?
Or does django have some magic in that it calls `to_representation` before saving?

Reply:
queries coming from the frontend should already be upgraded and the rest will eventually be updated by the backfill task. so i don't think that's necessary